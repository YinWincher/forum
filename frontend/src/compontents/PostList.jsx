import React from 'react';
import './postlist.css';
import {fetchDiscussion, getPostid, getPostMessage} from '../action/discussion'
import {Link} from 'react-router-dom'
const PostList = ({discussion,dispatch})=>{
    const {title,author,postid} = discussion;
    const handleLinkClick = (event)=>{
        const {postid} = event.currentTarget.dataset;
        dispatch(getPostid(postid))
        dispatch(fetchDiscussion('/api/discussion/'+postid))
        dispatch(getPostMessage(discussion));
    }
    return (
        <div className="border border-3 postlist-wrapper">
            <Link to={'/discussion/'+postid}>
            <div className='head' onClick={handleLinkClick} data-postid={postid}>
                <span className='title'>{title}</span>
                <span className='author'>{author}</span>
            </div>
            </Link>
        </div>
    );
}
export default PostList;