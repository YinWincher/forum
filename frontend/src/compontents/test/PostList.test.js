import PostList from '../PostList';
import React from 'react';
import {shallow} from  'enzyme';

const setup = ()=>{
    const props = {
        onSubmit :jest.fn(),
        discussion :{
            title:'test',
            author:'test',
            postid:1
        }
    };
    const wrapper = shallow(<PostList {...props}/>)
    return {
        props,
        wrapper
    }
}

describe('PublishArea',()=>{
    const {wrapper,props} = setup();

    //test ui render
    describe('ui render',()=>{
        test('render',()=>{
            expect(wrapper).toMatchSnapshot();
        })
    })

});