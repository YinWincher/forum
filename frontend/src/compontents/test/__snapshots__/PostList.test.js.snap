// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublishArea ui render render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PostList
    discussion={
        Object {
            "author": "test",
            "postid": 1,
            "title": "test",
          }
    }
    onSubmit={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        replace={false}
        to="/discussion/1"
>
        <div
                className="head"
                data-postid={1}
                onClick={[Function]}
        >
                <span
                        className="title"
                >
                        test
                </span>
                <span
                        className="author"
                >
                        test
                </span>
        </div>
</Link>,
      "className": "border border-3 postlist-wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="head"
          data-postid={1}
          onClick={[Function]}
>
          <span
                    className="title"
          >
                    test
          </span>
          <span
                    className="author"
          >
                    test
          </span>
</div>,
        "replace": false,
        "to": "/discussion/1",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="title"
>
              test
</span>,
            <span
              className="author"
>
              test
</span>,
          ],
          "className": "head",
          "data-postid": 1,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "test",
              "className": "title",
            },
            "ref": null,
            "rendered": "test",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "test",
              "className": "author",
            },
            "ref": null,
            "rendered": "test",
            "type": "span",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          replace={false}
          to="/discussion/1"
>
          <div
                    className="head"
                    data-postid={1}
                    onClick={[Function]}
          >
                    <span
                              className="title"
                    >
                              test
                    </span>
                    <span
                              className="author"
                    >
                              test
                    </span>
          </div>
</Link>,
        "className": "border border-3 postlist-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            className="head"
            data-postid={1}
            onClick={[Function]}
>
            <span
                        className="title"
            >
                        test
            </span>
            <span
                        className="author"
            >
                        test
            </span>
</div>,
          "replace": false,
          "to": "/discussion/1",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="title"
>
                test
</span>,
              <span
                className="author"
>
                test
</span>,
            ],
            "className": "head",
            "data-postid": 1,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "test",
                "className": "title",
              },
              "ref": null,
              "rendered": "test",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "test",
                "className": "author",
              },
              "ref": null,
              "rendered": "test",
              "type": "span",
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
